common/lib/common-utils/src/fluidError.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    IFluidError,
+    ITelemetryProperties,
+    ISensitiveDebugData,
+    isIFluidError,
+} from "@fluidframework/common-definitions";
+
+export abstract class FluidError extends Error implements IFluidError {
+    public abstract errorType: string;
+
+    private props: ITelemetryProperties = {};
+    private debugData: ISensitiveDebugData = {};
+    constructor(
+        message: string,
+        props: ITelemetryProperties = {},
+        debugData: ISensitiveDebugData = {},
```

Comment:
Can you please take a look how this is done in other places? I'm sure most other places use classification of properties, i.e. there is no 2 district sets (props & debugData), there is one set with annotations. And annotations support a lot of types.

For example here https://docs.microsoft.com/en-us/information-protection/develop/reference/class_mip_telemetryevent they have optional PII classification per property.
Can't find good documentation for Aria, but it does the same ([some deck ](https://microsoft.sharepoint.com/:p:/r/teams/BingRelevance/RelevanceInfrastructure/RnA/_layouts/15/doc2.aspx?sourcedoc=%7B8270CF6F-6DBB-49C6-9916-688B31BA8250%7D&file=Intro%20to%20Aria.pptx&action=edit&mobileredirect=true&DefaultItemOpen=1)I've found talks about it).

Note that even if we can't log actual payload, most systems will log hashed value of payload containing PII, allowing to at least count and do correlation analyzes (though it should be done in a way not to allow reversing process or identifying user via other means).

Reply:
This is good feedback.  I was taking a different approach, to separate safe stuff (`props`) from _potentially_ sensitive stuff (`debugData`). It's up to the host implementing the logger to inspect what's in `debugData` and decide how to handle it - we in FF don't have enough context to know. You can interpret this as a single classification of "potentially sensitive", with the expectation that the host will take it from there.

I will consider the alternative of adopting a granular classification system. One concern is whether it will be granular enough! As shown here, each and every field that gets added to `debugData` would be interpreted individually. The use of declaration merging would help ensure the semantics of each field are well-understood between the code adding the field and the code consuming it to log it or hash it or whatnot.